sdk/src/main/java/com/uid2/sdk/extensions/StringEx.kt

```diff
@@ -0,0 +1,16 @@
+package com.uid2.sdk.extensions
+
+import org.json.JSONObject
+
+/**
+ * Extension to parse a given String as JSON and convert to a Map. If parsing fails, e.g. the JSON
+ * is not well formed, then null will be returned.
+ */
+fun String.decodeJsonToMap(): Map<String, Any>? {
```

Comment:
Could be reduced to:
```
fun String.d(): Map<String, Any>? =
    kotlin.runCatching { JSONObject(this) }.getOrNull()?.let { json ->
        json.keys().asSequence().map { key -> key to json[key] }.toMap()
    }
```
But arguably harder to read....

Reply:
I'll leave for now, mainly because I think I'll be removing this function in a future PR.